packages/react-ui-core/src/Counter/Counter.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+
+export default class Counter extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      count: this.props.count,
+    }
+    this.onClick = this.onClick.bind(this)
+  }
+  onClick(type) {
+    const count = this.props.onClick(type, this.state.count)
+    this.setState({
+      count,
+    })
+  }
+  render() {
+    const { count } = this.state
+    const { theme, label, leftUnit, rightUnit, id } = this.props
+    return (
+      <div id={id} className={theme.container}>
+        <div className={theme.label}>
+          {label}
+        </div>
+        <div className={theme.componentContainer}>
+          <span
+            onClick={() => this.onClick('increment')}
```

Comment:
For increment and decrement we only care about the numeric value, not whether it went up or down. Lets modify the click event to just pass the number rather than a string: 

`onClick={() => this.onClick(this.state.count + 1)})`

rename onClick -> handleClick and make sure the props.onClick exists before calling it: 

```
handleClick(count) {
  if (this.props.onClick) {
    this.props.onClick(count)
  }
  this.setState(count)
}
```